packages/app/src/specs/SpecHeaderCloudDataTooltip.vue

```diff
@@ -0,0 +1,124 @@
+        <div>
+          <Auth
+            v-if="projectConnectionStatus === 'LOGGED_OUT'"
+            :gql="props.gql"
+            :show-retry="false"
+          />
+          <Button
+            v-else-if="projectConnectionStatus === 'NOT_CONNECTED'"
+            :prefix-icon="ConnectIcon"
+            prefix-icon-class="icon-dark-white icon-light-transparent"
+            :disabled="false"
+            @click="showConnectDialog = true"
+          >
+            {{ t("specPage.connectProjectButton") }}
+          </Button>
+          <Button
+            v-else-if="projectConnectionStatus === 'NOT_FOUND'"
+            :prefix-icon="ConnectIcon"
+            prefix-icon-class="icon-dark-white icon-light-transparent"
+            :disabled="false"
+            @click="showConnectDialog = true"
+          >
+            {{ t("specPage.reconnectProjectButton") }}
+          </Button>
+          <Button
+            v-else-if="projectConnectionStatus === 'UNAUTHORIZED'"
+            :prefix-icon="SendIcon"
+            prefix-icon-class="icon-dark-white icon-light-transparent"
+            :disabled="false"
+            @click="requestAccess"
+          >
+            {{ t("specPage.requestAccessButton") }}
+          </Button>
+        </div>
+      </div>
+    </template>
+  </Tooltip>
+  <CloudConnectModals
+    v-if="showConnectDialog"
+    :show="showConnectDialog"
+    :gql="props.gql"
+    @cancel="showConnectDialog = false"
+    @success="showConnectDialog = false"
+  />
+</template>
+
+<script setup lang="ts">
+import Auth from '@packages/frontend-shared/src/gql-components/Auth.vue'
+import CloudConnectModals from '../runs/modals/CloudConnectModals.vue'
+import Button from '@cy/components/Button.vue'
+import Tooltip from '@packages/frontend-shared/src/components/Tooltip.vue'
+import ConnectIcon from '~icons/cy/chain-link_x16.svg'
+import SendIcon from '~icons/cy/paper-airplane_x16.svg'
+import { RunsErrorRenderer_RequestAccessDocument, SpecHeaderCloudDataTooltipFragment } from '../generated/graphql'
+import { useI18n } from '@cy/i18n'
+import { computed, ref } from 'vue'
+import { gql, useMutation } from '@urql/vue'
+const { t } = useI18n()
+
+const props = defineProps<{
+  gql: SpecHeaderCloudDataTooltipFragment
+  headerText: string
+  connectedText: string
+  notConnectedText: string
+}>()
+
+const showConnectDialog = ref(false)
+
+gql`
+fragment SpecHeaderCloudDataTooltip on Query {
+  currentProject {
+    id
+    cloudProject{
+      __typename
+    }
+  }
+  ...Auth
+  ...CloudConnectModals
+}
+`
+
+const projectConnectionStatus = computed(() => {
+  if (!props.gql.cloudViewer) return 'LOGGED_OUT'
+
+  if (!props.gql.currentProject?.cloudProject?.__typename) return 'NOT_CONNECTED'
+
+  if (props.gql.currentProject?.cloudProject?.__typename === 'CloudProjectNotFound') return 'NOT_FOUND'
+
+  if (props.gql.currentProject?.cloudProject?.__typename === 'CloudProjectUnauthorized') return 'UNAUTHORIZED'
+
+  return 'CONNECTED'
+})
```

Comment:
Hmm, this seems correct but makes me wonder about a good way we could represent this in GraphQL, so you could just do:

```ts
currentProject {
  id
  cloudProjectStatus
}
```

Might take a look at that at some point, seems like a common thing we'll want to know across the app